version: '3'
services:
  mysqlsrv_chamados:
    image: mysql:5.7
    environment:
      MYSQL_ROOT_PASSWORD: "root"
      MYSQL_DATABASE: "chamados"
    ports:
      - "3397:3306"
    networks: 
      - netflix.case-network

  mysqlsrv_usuarios:
    image: mysql:5.7
    environment:
      MYSQL_ROOT_PASSWORD: "root"
      MYSQL_DATABASE: "usuario"
    ports:
      - "3398:3306"
    networks: 
      - netflix.case-network

  mysqlsrv_catalogos:
    image: mysql:5.7
    environment:
      MYSQL_ROOT_PASSWORD: "root"
      MYSQL_DATABASE: "catalogos"
    ports:
      - "3386:3306"
    networks: 
      - netflix.case-network

  chamados:
    build: 
      context: ./srv-chamados/
      dockerfile: ./Docker/Dockerfile
    ports:
      - "8080:8080"
    restart: on-failure
    depends_on:
      - configserver
      - mysqlsrv_chamados
    networks: 
      - netflix.case-network
    
  usuarios:
    build: 
      context: ./srv-usuario/
      dockerfile: ./Docker/Dockerfile
    ports:
      - "8181:8181"
    restart: on-failure
    depends_on:
      - configserver
      - mysqlsrv_usuarios
    networks: 
      - netflix.case-network

  catalogos:
    build: 
      context: ./srv-catalogo/
      dockerfile: ./Docker/Dockerfile
    ports:
      - "8383:8383"
    restart: on-failure
    depends_on:
      - configserver
      - mysqlsrv_catalogos
    networks: 
      - netflix.case-network
  
  gateway:
    build: 
      context: ./gateway/
      dockerfile: ./Docker/Dockerfile
    ports:
      - "8282:8282"
    restart: on-failure
    depends_on:
      - chamados
      - usuarios
      - catalogos
    networks: 
    - netflix.case-network
  
  configserver:
    build: 
      context: ./confsvr/
      dockerfile: ./Docker/Dockerfile
    ports:
      - "8888:8888"
    networks: 
      - netflix.case-network

networks: 
  netflix.case-network:
    driver: bridge